src/instance.ts

```diff
@@ -351,7 +351,12 @@ class Instance extends common.GrpcServiceObject {
     if (!name) {
       throw new Error('A name is required to access a Database object.');
     }
-    const key = name.split('/').pop();
+    // Only add an additional key for SessionPoolOptions if an options object with at least one value was passed in.
+    const optionsKey =
+      poolOptions && Object.keys(poolOptions).length > 0
+        ? '/' + JSON.stringify(poolOptions)
```

Comment:
Would the order of poolOptions affect the optionsKey? E.g., 

```javascript
{
   max: 10,
   min: 5,
}
```

```javascript
{
   min: 5,
   max: 10,
}
```

Will they lead to different optionsKey?

Reply:
That's a good point. Yes, that would lead to different option keys. I've changed it to a version that ignores the order of the properties.